kld/tests/smoke/start.rs

```diff
@@ -17,30 +17,51 @@ use kld::{
     },
     database::payment::PaymentStatus,
 };
-use test_utils::{bitcoin, cockroach, electrs, kld, poll, test_settings, TEST_ADDRESS};
+use tempfile::TempDir;
+use test_utils::{
+    poll, test_settings, BitcoinManager, CockroachManager, ElectrsManager, KldManager, TEST_ADDRESS,
+};
 use tokio::time::{sleep_until, Instant};
 
 #[tokio::test(flavor = "multi_thread")]
 pub async fn test_start() -> Result<()> {
-    let mut settings_0 = test_settings!("start");
-    let cockroach = cockroach!(settings_0);
-    let bitcoin = bitcoin!(settings_0);
+    let tmp_dir = TempDir::new()?;
+
+    let mut settings_0 = test_settings(&tmp_dir, "start");
+    let cockroach = CockroachManager::new(&tmp_dir, &mut settings_0).await?;
+    let bitcoin = BitcoinManager::new(&tmp_dir, &mut settings_0).await?;
     bitcoin
         .generate_blocks(START_N_BLOCKS, &Address::from_str(TEST_ADDRESS)?, false)
         .await?;
 
     settings_0.node_id = "start0".to_owned();
     settings_0.database_name = "start0".to_owned();
-    let electrs_0 = electrs!(&bitcoin, settings_0);
-    let kld_0 = kld!(&bitcoin, &cockroach, &electrs_0, settings_0);
+    let electrs_0 = ElectrsManager::new(&tmp_dir, &bitcoin, &mut settings_0).await?;
+    let kld_0 = KldManager::new(
+        &tmp_dir,
+        env!("CARGO_BIN_EXE_kld"),
+        &bitcoin,
+        &cockroach,
+        &electrs_0,
+        &mut settings_0,
+    )
+    .await?;
```

Comment:
The macros help by removing this boilerplate code but I can see how they are confusing. In the future we could have many more of these types of end to end test. We can find another way of hiding the setup code.